pymc/logprob/transforms.py

```diff
@@ -523,8 +539,64 @@ def measurable_sub_to_neg(fgraph, node):
     return [pt.add(minuend, pt.neg(subtrahend))]
 
 
+@node_rewriter([log1p, softplus, log1mexp, log2, log10])
+def measurable_special_log_to_log(fgraph, node):
+    """Convert log1p, log1mexp, softplus of `MeasurableVariable`s to log form."""
+    [inp] = node.inputs
+
+    if isinstance(node.op.scalar_op, Log1p):
+        return [pt.log(1 + inp)]
+    if isinstance(node.op.scalar_op, Softplus):
+        return [pt.log(1 + pt.exp(inp))]
+    if isinstance(node.op.scalar_op, Log1mexp):
+        return [pt.log(1 - pt.exp(pt.neg(inp)))]
+    if isinstance(node.op.scalar_op, Log2):
+        return [pt.log(inp) / pt.log(2)]
+    if isinstance(node.op.scalar_op, Log10):
+        return [pt.log(inp) / pt.log(10)]
+
+
+@node_rewriter([expm1, sigmoid])
+def measurable_special_exp_to_exp(fgraph, node):
+    """Convert expm1, sigmoid of `MeasurableVariable`s to xp form."""
+    [inp] = node.inputs
+    if isinstance(node.op.scalar_op, Expm1):
+        return [pt.exp(inp) - 1]
+    if isinstance(node.op.scalar_op, Sigmoid):
+        return [1 / (1 + pt.exp(-inp))]
+
+
+@node_rewriter([exp2, pow])
+def measurable_general_exp_to_exp(fgraph, node):
+    """Convert exp2 and any const^x of `MeasurableVariable`s to exp form."""
+    if len(node.inputs) > 1:
+        [const, inp] = node.inputs
+    else:
+        [inp] = node.inputs
+    if isinstance(node.op.scalar_op, Exp2):
+        return [pt.exp(pt.log(2) * inp)]
+    if isinstance(node.op.scalar_op, Pow) and isinstance(inp, pt.TensorVariable):
```

Comment:
Have to think about this carefully because stuff like x^2 needs the old special logic and for that the power must be fixed. Also this will fail with negative exponents

Reply:
I suggest we leave the new power stuff to a distinct PR

Reply:
Unless I'm mistaken this conserves the old logic so x^n still works as normal, because if x^2 then `inp` will be a scalar. But I agree it fails for const <= 0 which is more challenging. I'll revert back to just dealing exp2 and we can deal with the general power stuff another time.

Reply:
Sounds good. Also I was handwavy with "const". I meant a conditionally independent variable (either a constant or a variable that's valued).

Reply:
Ah I see, that makes it harder